Microsoft Edge 150 web platform release notes (Jul. 2026) (#3824)
MicrosoftDocs/edge-developer: microsoft-edge/web-platform/release-notes/150.md

This is useful for per-draw data, such as object indices, material indices, and transform values.

See also:
* [WebGPU API](https://developer.mozilla.org/docs/Web/API/WebGPU_API) at MDN.
* [WebGPU](https://gpuweb.github.io/gpuweb/) specification.
* [WebGPU Shading Language (WGSL)](https://gpuweb.github.io/gpuweb/wgsl/) specification.

<!-- ====================================================================== -->
## Origin trials

The following are origin trials for new experimental APIs that are available in Microsoft Edge.

Origin trials let you try experimental APIs on your own live website for a limited time.  To learn more about origin trials, see [Use origin trials in Microsoft Edge](../../origin-trials/index.md).

For the full list of available origin trials, see [Microsoft Edge Origin Trials](https://developer.microsoft.com/microsoft-edge/origin-trials/).

todo: compare table rows & their wording in [Microsoft Edge 149 web platform release notes (Jun. 2026)](./149.md)

> 

Author: ```suggestion
For the full list of available origin trials, see [Microsoft Edge Origin Trials](https://developer.microsoft.com/microsoft-edge/origin-trials/).

```

| Soft navigation heuristics | Exposes soft-navigation heuristics for collecting performance metrics in single-page apps. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/ec952482-49f0-4980-a451-13ea3f9a7220) |
| WebNN | Build and execute hardware-accelerated machine-learning models directly in your web app. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/19857284-cf52-484b-8b09-8ca50ac9dccb) |
| URL and eval hashes in CSP script-src | Introduces url- and eval- hashes for `script-src` CSP directives, replacing hostname-based allowlists and `unsafe-eval`. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/6bd50a0d-6f30-425f-beba-034be41bb013) |
| Web Install API | Allows a website to install another website as a web app by using `navigator.install()`. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/bcf7d28f-6bdf-45d8-9207-63c97a042407) |

> | <!-- new -->`OpaqueRange` | Enables creating live ranges over text inside `<input>` and `<textarea>` elements that automatically update as the user edits.  These ranges support geometry methods such as `getBoundingClientRect()` and the CSS Custom Highlight API, which enables use cases such as caret-positioned popups and inline spell-check highlighting directly on form controls, without cloning elements or exposing internal DOM structure. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/5f4620e8-2969-4579-a5c6-304b8fae7200) |

Author: ```suggestion
| `OpaqueRange` | Enables creating live ranges over text inside `<input>` and `<textarea>` elements that automatically update as the user edits.  These ranges support geometry methods such as `getBoundingClientRect()` and the CSS Custom Highlight API, which enables use cases such as caret-positioned popups and inline spell-check highlighting directly on form controls, without cloning elements or exposing internal DOM structure. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/5f4620e8-2969-4579-a5c6-304b8fae7200) |
```

| WebAudio Configurable Render Quantum | Lets you specify a custom render quantum size when creating an `AudioContext` or `OfflineAudioContext`. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/d5574838-d416-4242-81a8-7833c107b028) |
| Prerender activation by form submission | Allows prerender rules in the Speculation Rules API to be activated by form submissions. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/63420781-6a9d-4b23-a827-69ebe66fb631) |
| CPU Performance API | Exposes information about the user's device capabilities, for use with the Compute Pressure API. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/505dbb22-2d7f-4c91-9a2b-3913345a956e) |
| Connection allowlists | Restricts connections from a document or worker to a server-distributed allowlist of endpoints. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/a20a4374-39fd-445f-9348-f6c8737a45de) |

> | <!-- new -->`focus-without-user-activation` Permissions Policy | Gives an embedder control over programmatic focus from embedded content via the `focus-without-user-activation` permissions policy.  When the policy is denied for a frame, programmatic focus calls (`element.focus()`, `autofocus`, `window.focus()`, `dialog.showModal()`, and popover focusing) are blocked unless triggered by user activation, or focus is being delegated. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/e5a1d64f-b04b-453f-9550-6d12c1f6bc99) |

Author: ```suggestion
| `focus-without-user-activation` permissions policy | Gives an embedder control over programmatic focus from embedded content via the `focus-without-user-activation` permissions policy.  When the policy is denied for a frame, programmatic focus calls (`element.focus()`, `autofocus`, `window.focus()`, `dialog.showModal()`, and popover focusing) are blocked unless triggered by user activation, or focus is being delegated. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/e5a1d64f-b04b-453f-9550-6d12c1f6bc99) |
```

| Separate style and layout durations in the Long Animation Frame API | Adds `styleDuration`, `forcedStyleDuration`, `layoutDuration`, and `forcedLayoutDuration` properties for deeper CSS performance analysis. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/349e23d8-799a-4ac2-b45a-ffb91e09b4ce) |
| Prompt API sampling parameters | The `topK` and `temperature` sampling parameters let you optimize model behavior per language model session. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/32d0cd0f-611d-411e-b48e-cfd8edfdbf73) |
| `<install>` HTML element | Declaratively install other websites as web apps by using the `<install>` element. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/b2eb5c7f-7df0-4ad9-950e-2fbd2ebc08ec) |
| Declarative CSS Module Scripts | Shares declarative stylesheets with shadow roots, including declarative shadow roots, using inline style modules. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/966b04dd-3da4-43f4-a6a0-66a785f129f4) |

> | <!-- new -->Gamepad Raw Input Event | Instead of relying on frequent polling via `navigator.getGamepads()`, web pages can listen for `gamepadrawinputchanged` events.  These events are fired whenever changes are detected between input frames from the same device. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/4ecdb530-835b-447d-b27e-0b3be8e0412b) |

Author: ```suggestion
| Gamepad raw input event | Instead of relying on frequent polling via `navigator.getGamepads()`, web pages can listen for `gamepadrawinputchanged` events.  These events are fired whenever changes are detected between input frames from the same device. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/4ecdb530-835b-447d-b27e-0b3be8e0412b) |
```

| Autofill Event | Detects when browser autofill updates form controls, for adapting custom UI and validation. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/43534059-5696-4eaf-8bbd-eba4c4f86e88) |
| WebAssembly Custom Descriptors V2 | Stores data associated with source-level types more efficiently, in custom descriptor objects. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/9c75e7f3-49cb-43de-9745-7458ff9c3689) |

> | <!-- new -->WebMCP | Enables your site to register tools for an in-browser agent to complete tasks on behalf of a user. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/0b76fe60-b266-458e-a285-04e375c0c31a) |

Author: ```suggestion
| WebMCP | Enables your site to register tools for an in-browser agent to complete tasks on behalf of a user. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/0b76fe60-b266-458e-a285-04e375c0c31a) |
```

   * [Web Speech API: on-device recognition quality](#web-speech-api-on-device-recognition-quality)
   * [WebGPU: Immediates](#webgpu-immediates)
* [Origin trials](#origin-trials)

<!-- ====================================================================== -->
## Microsoft Edge DevTools

See [What's new in Microsoft Edge DevTools](../../devtools/whats-new/index.md).
<!-- todo: when exists, use instead:
See [What's new in DevTools (Microsoft Edge 150)](../../devtools/whats-new/150.md).
-->

<!-- ====================================================================== -->
## WebView2

See [Release notes for the WebView2 SDK](../../webview2/release-notes/index.md).
<!-- todo: when exists, use instead:
See [Release SDK 1.0.nnnn.nn, for Runtime nnn](../../webview2/release-notes/index.md#10nnnnnn) (Jul 2026) in _Release notes for the WebView2 SDK_.
-->

<!-- ====================================================================== -->
## CSS features

The following new Cascading Style Sheets (CSS) features are included in Microsoft Edge.

#### `AccentColor` and `AccentColorText` system colors

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->
> 
> The `AccentColor` and `AccentColorText` system colors can now be used as CSS color values to access the system accent color that's specified on the user's device.

Author: ```suggestion
#### `AccentColor` and `AccentColorText` system colors

The `AccentColor` and `AccentColorText` system colors can now be used as CSS color values to access the system accent color that's specified on the user's device.
```

See also:
* [\<system-color\>](https://developer.mozilla.org/docs/Web/CSS/system-color) at MDN.
* [system Colors](https://drafts.csswg.org/css-color/#css-system-colors) in _CSS Color Module Level 4_.

<!-- ------------------------------ -->
#### Allow optional rounding parameter for `polygon()`

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->
> 
> Allows an optional `round` parameter in the `polygon()` CSS shape function.  You can specify a length value to round polygon corners, without manually computing bezier curves.

Author: ```suggestion
#### Allow optional rounding parameter for `polygon()`

Allows an optional `round` parameter in the `polygon()` CSS shape function.  You can specify a length value to round polygon corners, without manually computing bezier curves.
```

For example, to create a rounded triangle: `polygon(round 50px, 50% 0%, 0% 100%, 100% 100%)`

See also:
* [CSS polygon() round playground](https://codepen.io/yisi/pen/NPRLEQN) - Interactive demo.
* [[css-shapes] Allow optional rounding parameter for polygon()  #9843](https://github.com/w3c/csswg-drafts/issues/9843) - CSS Working Group issue.

<!-- ------------------------------ -->
#### Animatable zoom

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->
> 
> The CSS `zoom` property can be animated, and interpolates as a `<number>`.  You can transition and animate zoom to smoothly scale elements and their layout, complementing existing transform-based scaling.

Author: ```suggestion
#### Animatable zoom

The CSS `zoom` property can be animated, and interpolates as a `<number>`.  You can transition and animate zoom to smoothly scale elements and their layout, complementing existing transform-based scaling.
```

See also:
* [`zoom` CSS property](https://developer.mozilla.org/docs/Web/CSS/Reference/Properties/zoom) at MDN.

<!-- ------------------------------ -->
#### CSS URL request modifiers

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->
> 
> The following CSS functions can be used with the CSS `url()` function to control the fetch behavior of the referenced resource:

Author: ```suggestion
#### CSS URL request modifiers

The following CSS functions can be used with the CSS `url()` function to control the fetch behavior of the referenced resource:
```

<!-- ------------------------------ -->
#### CSS `image(<color>)` function

The CSS `image()` function enables you to easily generate a solid-color image from any color.

For example, to create a solid red background-image, use `background: image(red);`.

See also:
* [\<image\> CSS type](https://developer.mozilla.org/docs/Web/CSS/Reference/Values/image) at MDN.
* [Image Fallbacks and Annotations: the `image()` notation](https://drafts.csswg.org/css-images-4/#image-notation) in _CSS Images Module Level 4_.

<!-- ------------------------------ -->
#### CSS `light-dark()` with image values

The CSS `light-dark()` function now accepts image values (`url()`, `image-set()`, `none`).

This makes it possible to automatically switch between images based on the user's preferred color scheme.

You can use the `light-dark()` function with image values in the following properties:
* `background-image`
* `list-style-image`
* `border-image-source`
* `cursor`
* `content`

See also:
* [`light-dark()` CSS function](https://developer.mozilla.org/docs/Web/CSS/Reference/Values/color_value/light-dark) at MDN.

<!-- ------------------------------ -->
#### Comma-separated container queries

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->
> 
> The `@container` rule now supports multiple container queries, separated by commas.  The `@container` rule applies if at least one of the queries matches.

Author: ```suggestion
#### Comma-separated container queries

The `@container` rule now supports multiple container queries, separated by commas.  The `@container` rule applies if at least one of the queries matches.
```

Example:

`@container --name1 not-supported(--foo: bar), --name2 (width > 600px) {}`

See also:
* [CSS container querie](https://developer.mozilla.org/docs/Web/CSS/Guides/Containment/Container_queries) at MDN.

<!-- ------------------------------ -->
#### Expose unprintable areas via CSS

Use the CSS `page-margin-safety` property to avoid trying to print to unprintable areas of a sheet of paper.

A printer usually has a small area at each of the four edges of a sheet of paper that the printer is not capable of marking reliably, usually due to the printer's paper handling mechanism.

The default page margins are expected to be bigger than these paper-edge areas, but if you set margins in your app, and want to add `@page` margin boxes (such as for custom headers and footers), the `page-margin-safety` property provides a way of detecting where it's safe to print.

See also:
* [Staying within the printable area: the `page-margin-safety` property](https://drafts.csswg.org/css-page-3/#page-margin-safety) in _CSS Paged Media Module Level 3_.

<!-- ------------------------------ -->
#### Media element pseudo-classes

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->
> 
> The following CSS pseudo-classes match `<audio>` and `<video>` elements based on their state:

Author: ```suggestion
#### Media element pseudo-classes

The following CSS pseudo-classes match `<audio>` and `<video>` elements based on their state:
```

* `:playing`
* `:paused`
* `:seeking`
* `:buffering`
* `:stalled`
* `:muted`
* `:volume-locked`

See also:
* [`:playing` CSS pseudo-class](https://developer.mozilla.org/docs/Web/CSS/Reference/Selectors/:playing) at MDN.
* [`:paused` CSS pseudo-class](https://developer.mozilla.org/docs/Web/CSS/Reference/Selectors/:paused) at MDN.
* [`:seeking` CSS pseudo-class](https://developer.mozilla.org/docs/Web/CSS/Reference/Selectors/:seeking) at MDN.
* [`:buffering` CSS pseudo-class](https://developer.mozilla.org/docs/Web/CSS/Reference/Selectors/:buffering) at MDN.
* [`:stalled` CSS pseudo-class](https://developer.mozilla.org/docs/Web/CSS/Reference/Selectors/:stalled) at MDN.
* [`:muted` CSS pseudo-class](https://developer.mozilla.org/docs/Web/CSS/Reference/Selectors/:muted) at MDN.
* [`:volume-locked` CSS pseudo-class](https://developer.mozilla.org/docs/Web/CSS/Reference/Selectors/:volume-locked) at MDN.

<!-- ------------------------------ -->
#### CSS `alpha()` function

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->
> 
> The CSS `alpha()` function creates a new color by applying a specified alpha value to an existing color.

Author: ```suggestion
#### CSS `alpha()` function

The CSS `alpha()` function creates a new color by applying a specified alpha value to an existing color.
```

For example, `alpha(from red / 0.5)` creates a semi-transparent red color with 50% opacity.

See also:
* [Relative Alpha Colors](https://drafts.csswg.org/css-color-5/#relative-alpha) in _CSS Color Module Level 5_.

<!-- ------------------------------ -->
#### `named-feature()` function for CSS `@supports`

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->
> 
> The `named-feature()` function allows CSS `@supports` rules to query for a small set of specific named features that are not possible to test for by using other `@supports` mechanisms, but which are valuable to test for.

Author: ```suggestion
#### `named-feature()` function for CSS `@supports`

The `named-feature()` function allows CSS `@supports` rules to query for a small set of specific named features that are not possible to test for by using other `@supports` mechanisms, but which are valuable to test for.
```

See also:
* [Extensions to the `@supports` rule](https://drafts.csswg.org/css-conditional-5/#at-supports-ext) in _CSS Conditional Rules Module Level 5_.

<!-- ------------------------------ -->
#### `overscroll-behavior: chain`

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->
> 
> The `overscroll-behavior` CSS property now supports the `chain` value.

Author: ```suggestion
#### `overscroll-behavior: chain`

The `overscroll-behavior` CSS property now supports the `chain` value.
```

This property affects two independent effects:
* Scroll propagation.
* Local border effect (such as overscroll stretch).

Previously, `overscroll-behavior` had three values:
* `none`: No propagation, no local effect.
* `auto`: Propagation, local effect.
* `contain`: No propagation, local effect.

This feature adds a new value to complete the set:
* `chain`: Propagation, no local effect.

See also:
* [`overscroll-behavior` CSS property](https://developer.mozilla.org/docs/Web/CSS/Reference/Properties/overscroll-behavior) at MDN.
* [Overscroll Behavior Properties](https://drafts.csswg.org/css-overscroll-1/#overscroll-behavior-properties) in _CSS Overscroll Behavior Module Level 1_.

<!-- ------------------------------ -->
#### `flex-wrap:balance`

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->
> 
> `flex-wrap:balance` allows you to distribute content between each `flex-line`, so that the layout appears more balanced.

Author: ```suggestion
#### `flex-wrap:balance`

`flex-wrap:balance` allows you to distribute content between each `flex-line`, so that the layout appears more balanced.
```

See also:
* [Popover API](https://developer.mozilla.org/docs/Web/API/Popover_API) at MDN.
* [`popover` HTML global attribute](https://developer.mozilla.org/docs/Web/HTML/Reference/Global_attributes/popover) at MDN.

<!-- ------------------------------ -->
#### Updating all `<selectedcontent>` element in a customizable `<select>`

When multiple `<selectedcontent>` elements are defined in a customizable `<select>` element, all of them are now be kept up to date with the current selected option, instead of only the first one in DOM order.

See also:
* [Customizable select elements](https://developer.mozilla.org/docs/Learn_web_development/Extensions/Forms/Customizable_select) at MDN.
* [\<selectedcontent\> HTML selected option display element](https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/selectedcontent) at MDN.

<!-- ------------------------------ -->
#### Streaming out-of-order HTML

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->
> 
> Use the `<template for>` HTML element and the `<?start>` and `<?end>` processing instruction ranges to update existing parts of the document without needing to use JavaScript.

Author: ```suggestion
#### Streaming out-of-order HTML

Use the `<template for>` HTML element and the `<?start>` and `<?end>` processing instruction ranges to update existing parts of the document without needing to use JavaScript.
```

<!-- ------------------------------ -->
#### Expose the `FontFaceSet` interface globally

Previously, the `FontFaceSet` interface was not exposed as a global property.  Chromium's interface definition language (IDL) for `FontFaceSet` previously incorrectly applied `LegacyNoInterfaceObject` to `FontFaceSet`, hiding it as a global property.

This change aligns Chromium with the CSS Font Loading spec, and with Safari and Firefox.

See also:
* [FontFaceSet](https://developer.mozilla.org/docs/Web/API/FontFaceSet) at MDN.

<!-- ------------------------------ -->
#### Disable SVG filters on cross-origin iframes and plugins

SVG filters are no longer applied to the following:
* Cross-origin or restricted iframes (such as sandboxed iframes).
* Embedded plugins (such as PDFs).

This prevents potential security issues from cross-origin content being processed through SVG filter effects.

See also:
* [SVG filters](https://developer.mozilla.org/docs/Web/SVG/Guides/SVG_filters) at MDN.
* [`<iframe>` HTML inline frame element](https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/iframe) at MDN.

<!-- ------------------------------ -->
#### `MediaStreamTrackProcessor` frame counters 

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->
> 
> The `discardedFrames` and `totalFrames` attributes are now available on the `MediaStreamTrackProcessor` interface.

Author: ```suggestion
#### `MediaStreamTrackProcessor` frame counters 

The `discardedFrames` and `totalFrames` attributes are now available on the `MediaStreamTrackProcessor` interface.
```

This aligns Edge with the HTML Standard and improves security by isolating these workers from same-origin state, such as `BroadcastChannel`, `localStorage`, and `IndexedDB`.

To preserve isolation boundaries, these workers remain in the same storage partition as their creator (for example, by top-level site or nonce).

See also:
* [RFC 2397: The "data" URL scheme](https://datatracker.ietf.org/doc/html/rfc2397).
* [Step 3 of Script settings for workers](https://html.spec.whatwg.org/multipage/workers.html#script-settings-for-workers) in _HTML Living Standard_.

<!-- ------------------------------ -->
#### PWA origin migration

Web App Origin Migration enables a Progressive Web App (PWA) to migrate to a new origin while preserving trust, install state, and applicable permissions.

This helps teams safely move apps across domains without forcing users to reinstall.

See also:
* [Web App Manifest](https://developer.mozilla.org/docs/Web/Manifest) at MDN.

<!-- ------------------------------ -->
#### Get notified when the `scrollBy` and `scrollTo` methods complete

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->
> 
> Programmatic scrolling methods, such as `scrollBy` and `scrollTo`, now return a `Promise` object that resolves when the scroll completes.  Use this promise to run code after a smooth-scroll finishes, without relying on a timer or scroll-event polling.

Author: ```suggestion
#### Get notified when the `scrollBy` and `scrollTo` methods complete

Programmatic scrolling methods, such as `scrollBy` and `scrollTo`, now return a `Promise` object that resolves when the scroll completes.  Use this promise to run code after a smooth-scroll finishes, without relying on a timer or scroll-event polling.
```

See also:
* [Window: scrollBy() method](https://developer.mozilla.org/docs/Web/API/Window/scrollBy) at MDN.
* [Window: scrollTo() method](https://developer.mozilla.org/docs/Web/API/Window/scrollTo) at MDN.

<!-- ------------------------------ -->
#### Web Speech API: on-device recognition quality

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->
> 
> The `SpeechRecognition` interface now supports the `quality` property.

Author: ```suggestion
#### Web Speech API: on-device recognition quality

The `SpeechRecognition` interface now supports the `quality` property.
```

todo: compare table rows & their wording in [Microsoft Edge 149 web platform release notes (Jun. 2026)](./149.md)

| Name | Description | Register |
|---|---|---|
| Prompt API | Prompts a built-in language model from your website or extension JavaScript code.  See also [Prompt API](../prompt-api.md). | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/b7d35247-b855-4b08-b237-89e7a5056117) |
| WebAssembly custom descriptors | Stores data associated with source-level types more efficiently, in custom descriptor objects. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/f1227421-7d96-4380-b588-cbe9806aa7a3) |
| Soft navigation heuristics | Exposes soft-navigation heuristics for collecting performance metrics in single-page apps. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/ec952482-49f0-4980-a451-13ea3f9a7220) |
| WebNN | Build and execute hardware-accelerated machine-learning models directly in your web app. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/19857284-cf52-484b-8b09-8ca50ac9dccb) |

> | `focusgroup` HTML attribute | Standardizes keyboard navigation for composite widgets such as toolbars, tabs, menus, and radio groups. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/6d4b86e7-230c-4d23-ab2b-ee8d05f18702) |
> | URL and eval hashes in CSP script-src | Introduces url- and eval- hashes for `script-src` CSP directives, replacing hostname-based allowlists and `unsafe-eval`. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/6bd50a0d-6f30-425f-beba-034be41bb013) |

Author: ```suggestion
| WebNN | Build and execute hardware-accelerated machine-learning models directly in your web app. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/19857284-cf52-484b-8b09-8ca50ac9dccb) |
| URL and eval hashes in CSP script-src | Introduces url- and eval- hashes for `script-src` CSP directives, replacing hostname-based allowlists and `unsafe-eval`. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/6bd50a0d-6f30-425f-beba-034be41bb013) |
```

**Detailed contents:**
* [Microsoft Edge DevTools](#microsoft-edge-devtools)
* [WebView2](#webview2)
* [CSS features](#css-features)
   * [`AccentColor` and `AccentColorText` system colors](#accentcolor-and-accentcolortext-system-colors)
   * [Allow optional rounding parameter for `polygon()`](#allow-optional-rounding-parameter-for-polygon)
   * [Animatable zoom](#animatable-zoom)
   * [CSS URL request modifiers](#css-url-request-modifiers)
   * [CSS `text-fit` property](#css-text-fit-property)
   * [CSS `background-clip: border-area`](#css-background-clip-border-area)
   * [CSS `image(<color>)` function](#css-imagecolor-function)
   * [CSS `light-dark()` with image values](#css-light-dark-with-image-values)
   * [Comma-separated container queries](#comma-separated-container-queries)
   * [Expose unprintable areas via CSS](#expose-unprintable-areas-via-css)
   * [Media element pseudo-classes](#media-element-pseudo-classes)
   * [CSS `alpha()` function](#css-alpha-function)
   * [`named-feature()` function for CSS @supports](#named-feature-function-for-css-supports)
   * [`overscroll-behavior: chain`](#overscroll-behavior-chain)
   * [`flex-wrap:balance`](#flex-wrapbalance)

>    * [CSS `frame-sizing` property to responsively size `<iframe>` elements](#css-frame-sizing-property-to-responsively-size-iframe-elements)
> * [HTML features](#html-features)

Author: ```suggestion
   * [`flex-wrap:balance`](#flex-wrapbalance)
* [HTML features](#html-features)
```

   * [`focusgroup` attribute](#focusgroup-attribute)
   * [`popover="hint"` attribute behavior changes](#popoverhint-attribute-behavior-changes)
   * [Updating all `<selectedcontent>` element in a customizable `<select>`](#updating-all-selectedcontent-element-in-a-customizable-select)
   * [Streaming out-of-order HTML](#streaming-out-of-order-html)
* [Web APIs](#web-apis)
   * [Expose the `FontFaceSet` interface globally](#expose-the-fontfaceset-interface-globally)
   * [Disable SVG filters on cross-origin iframes and plugins](#disable-svg-filters-on-cross-origin-iframes-and-plugins)

>    * [IndexedDB: SQLite backend](#indexeddb-sqlite-backend)
>    * [`MediaStreamTrackProcessor` frame counters](#mediastreamtrackprocessor-frame-counters)

Author: ```suggestion
   * [Disable SVG filters on cross-origin iframes and plugins](#disable-svg-filters-on-cross-origin-iframes-and-plugins)
   * [`MediaStreamTrackProcessor` frame counters](#mediastreamtrackprocessor-frame-counters)
```

<!-- ------------------------------ -->
#### `flex-wrap:balance`

<!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

`flex-wrap:balance` allows you to distribute content between each `flex-line`, so that the layout appears more balanced.

This is similar to `text-wrap:balance`.

See also:
* [`flex-wrap` CSS property](https://developer.mozilla.org/docs/Web/CSS/Reference/Properties/flex-wrap) at MDN.
* [Flex Line Wrapping: the `flex-wrap` property](https://drafts.csswg.org/css-flexbox-2/#flex-wrap-property) in _CSS Flexible Box Layout Module Level 2_.

<!-- ------------------------------ -->
#### CSS `frame-sizing` property to responsively size `<iframe>` elements

<!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Using the CSS `frame-sizing` property, you can seamlessly render an iframe's content without scrollbars, by automatically resizing the iframe to fit its content.

See also:
* [`<iframe>` HTML inline frame element](https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/iframe) at MDN.

> * [Responsively-sized iframes: the `frame-sizing` property](https://drafts.csswg.org/css-sizing-4/#responsive-iframes) in _CSS Box Sizing Module Level 4_.
> 
> 
> <!-- ====================================================================== -->

Author: ```suggestion
* [Flex Line Wrapping: the `flex-wrap` property](https://drafts.csswg.org/css-flexbox-2/#flex-wrap-property) in _CSS Flexible Box Layout Module Level 2_.

<!-- ====================================================================== -->
```

<!-- ------------------------------ -->
#### Disable SVG filters on cross-origin iframes and plugins

SVG filters are no longer applied to the following:
* Cross-origin or restricted iframes (such as sandboxed iframes).
* Embedded plugins (such as PDFs).

This prevents potential security issues from cross-origin content being processed through SVG filter effects.

See also:
* [SVG filters](https://developer.mozilla.org/docs/Web/SVG/Guides/SVG_filters) at MDN.
* [`<iframe>` HTML inline frame element](https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/iframe) at MDN.

<!-- ------------------------------ -->
#### IndexedDB: SQLite backend

<!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

In Edge, the implementation of IndexedDB now uses SQLite to improve reliability and, to a lesser extent, performance.

With this change, the first time a user visits a site, or after clearing site data, new data is stored in the IndexedDB backend that uses SQLite.  Existing data stores will be migrated in a future release.

See also:

> * [IndexedDB API](https://developer.mozilla.org/docs/Web/API/IndexedDB_API) at MDN.
> 
> 
> <!-- ------------------------------ -->

Author: ```suggestion
* [`<iframe>` HTML inline frame element](https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/iframe) at MDN.

<!-- ------------------------------ -->
```

todo: compare table rows & their wording in [Microsoft Edge 149 web platform release notes (Jun. 2026)](./149.md)

| Name | Description | Register |
|---|---|---|
| Prompt API | Prompts a built-in language model from your website or extension JavaScript code.  See also [Prompt API](../prompt-api.md). | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/b7d35247-b855-4b08-b237-89e7a5056117) |
| WebAssembly custom descriptors | Stores data associated with source-level types more efficiently, in custom descriptor objects. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/f1227421-7d96-4380-b588-cbe9806aa7a3) |
| Soft navigation heuristics | Exposes soft-navigation heuristics for collecting performance metrics in single-page apps. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/ec952482-49f0-4980-a451-13ea3f9a7220) |
| WebNN | Build and execute hardware-accelerated machine-learning models directly in your web app. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/19857284-cf52-484b-8b09-8ca50ac9dccb) |

> | `focusgroup` HTML attribute | Standardizes keyboard navigation for composite widgets such as toolbars, tabs, menus, and radio groups. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/6d4b86e7-230c-4d23-ab2b-ee8d05f18702) |

Reviewer: If focusgroup is confirmed to be shipping with 150 (see regular section https://review.learn.microsoft.com/en-us/microsoft-edge/web-platform/release-notes/150?branch=pr-en-us-3824#focusgroup-attribute above), then remove this origin trial row.

| URL and eval hashes in CSP script-src | Introduces url- and eval- hashes for `script-src` CSP directives, replacing hostname-based allowlists and `unsafe-eval`. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/6bd50a0d-6f30-425f-beba-034be41bb013) |
| Web Install API | Allows a website to install another website as a web app by using `navigator.install()`. | [Register](https://developer.microsoft.com/microsoft-edge/origin-trials/trials/bcf7d28f-6bdf-45d8-9207-63c97a042407) |

> | <!-- new -->`OpaqueRange` | Enables creating live ranges over text inside `<input>` and `<textarea>` elements that automatically update as the user edits.  These ranges support geometry methods such as `getBoundingClientRect()` and the CSS Custom Highlight API, which enables use cases such as caret-positioned popups and inline spell-check highlighting directly on form controls, without cloning elements or exposing internal DOM structure. | [Register](https://developer.microsoft.com/en-us/microsoft-edge/origin-trials/trials/5f4620e8-2969-4579-a5c6-304b8fae7200) |

Reviewer: For the OT table, copied rows from 149.md based on the Register GUID.  
In rows that are not in 149.md, added a `<!-- new -->` flag, which can be removed.
@captainbrosset pls review wording of 6 OT rows marked `<!-- new -->`.

This enables gradient borders without `border-image`.

See also:
* [`background-clip` CSS property](https://developer.mozilla.org/docs/Web/CSS/Reference/Properties/background-clip) at MDN.

<!-- ------------------------------ -->
#### CSS `image(<color>)` function

The CSS `image()` function allows you to easily generate a solid-color image from any color.

For example, to create a solid red background-image, use `background: image(red);`.

See also:
* [\<image\> CSS type](https://developer.mozilla.org/docs/Web/CSS/Reference/Values/image) at MDN.
* [Image Fallbacks and Annotations: the `image()` notation](https://drafts.csswg.org/css-images-4/#image-notation) in _CSS Images Module Level 4_.

<!-- ------------------------------ -->
#### CSS `light-dark()` with image values

The CSS `light-dark()` function now accepts image values (`url()`, `image-set()`, `none`).

This makes it possible to automatically switch between images based on the user's preferred color scheme.

> You can use the `light-dark()` function with image values in the following porperties:

Reviewer: ```suggestion
You can use the `light-dark()` function with image values in the following properties:
```

   * [`focusgroup` attribute](#focusgroup-attribute)
   * [`popover=hint` attribute behavior changes](#popoverhint-attribute-behavior-changes)
   * [Updating all `<selectedcontent>` element in a customizable `<select>`](#updating-all-selectedcontent-element-in-a-customizable-select)
   * [Streaming out-of-order HTML](#streaming-out-of-order-html)
* [Web APIs](#web-apis)
   * [Expose the `FontFaceSet` interface globally](#expose-the-fontfaceset-interface-globally)
   * [Disable SVG filters on cross-origin iframes and plugins](#disable-svg-filters-on-cross-origin-iframes-and-plugins)
   * [IndexedDB: SQLite backend](#indexeddb-sqlite-backend)
   * [`MediaStreamTrackProcessor` frame counters](#mediastreamtrackprocessor-frame-counters)
   * [Opaque origin for `data:` URLs](#opaque-origin-for-data-urls)
   * [PWA origin migration](#pwa-origin-migration)
   * [Get notified when the `scrollBy` and `scrollTo` methods complete](#get-notified-when-the-scrollby-and-scrollto-methods-complete)
   * [Web Speech API: on-device recognition quality](#web-speech-api-on-device-recognition-quality)
   * [WebGPU: Immediates](#webgpu-immediates)
* [Origin trials](#origin-trials)

> 

Author: ```suggestion

```

<!-- ====================================================================== -->
## Microsoft Edge DevTools

See [What's new in Microsoft Edge DevTools](../../devtools/whats-new/index.md).
<!-- todo: when exists, use instead:
See [What's new in DevTools (Microsoft Edge 150)](../../devtools/whats-new/150.md).
-->

<!-- ====================================================================== -->
## WebView2

See [Release notes for the WebView2 SDK](../../webview2/release-notes/index.md).
<!-- todo: when exists, use instead:
See [Release SDK 1.0.nnnn.nn, for Runtime nnn](../../webview2/release-notes/index.md#10nnnnnn) (Jul 2026) in _Release notes for the WebView2 SDK_.
-->

<!-- ====================================================================== -->
## CSS features

The following new Cascading Style Sheets (CSS) features are included in Microsoft Edge.

#### `AccentColor` and `AccentColorText` system colors

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Author: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging

The `AccentColor` and `AccentColorText` system colors can now be used as CSS color values to access the system accent color that's specified on the user's device.

See also:
* [\<system-color\>](https://developer.mozilla.org/docs/Web/CSS/system-color) at MDN.
* [system Colors](https://drafts.csswg.org/css-color/#css-system-colors) in _CSS Color Module Level 4_.

<!-- ------------------------------ -->
#### Allow optional rounding parameter for `polygon()`

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Author: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging

Allows an optional `round` parameter in the `polygon()` CSS shape function.  You can specify a length value to round polygon corners, without manually computing bezier curves.

For example, to create a rounded triangle: `polygon(round 50px, 50% 0%, 0% 100%, 100% 100%)`

See also:
* [CSS polygon() round playground](https://codepen.io/yisi/pen/NPRLEQN) - Interactive demo.
* [[css-shapes] Allow optional rounding parameter for polygon()  #9843](https://github.com/w3c/csswg-drafts/issues/9843) - CSS Working Group issue.

<!-- ------------------------------ -->
#### Animatable zoom

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Author: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging

The CSS `zoom` property can be animated, and interpolates as a `<number>`.  You can transition and animate zoom to smoothly scale elements and their layout, complementing existing transform-based scaling.

See also:
* [`zoom` CSS property](https://developer.mozilla.org/docs/Web/CSS/Reference/Properties/zoom) at MDN.

<!-- ------------------------------ -->
#### CSS URL request modifiers

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Author: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging

<!-- ------------------------------ -->
#### CSS `image(<color>)` function

The CSS `image()` function allows you to easily generate a solid-color image from any color.

For example, to create a solid red background-image, use `background: image(red);`.

See also:
* [\<image\> CSS type](https://developer.mozilla.org/docs/Web/CSS/Reference/Values/image) at MDN.
* [Image Fallbacks and Annotations: the `image()` notation](https://drafts.csswg.org/css-images-4/#image-notation) in _CSS Images Module Level 4_.

<!-- ------------------------------ -->
#### CSS `light-dark()` with image values

The CSS `light-dark()` function now accepts image values (`url()`, `image-set()`, `none`).

This makes it possible to automatically switch between images based on the user's preferred color scheme.

You can use the `light-dark()` function with image values in the following porperties:
* `background-image`
* `list-style-image`
* `border-image-source`
* `cursor`
* `content`

See also:
* [`light-dark()` CSS function](https://developer.mozilla.org/docs/Web/CSS/Reference/Values/color_value/light-dark) at MDN.

<!-- ------------------------------ -->
#### Comma-separated container queries

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Author: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging

Example:

`@container --name1 not-supported(--foo: bar), --name2 (width > 600px) {}`

See also:
* [CSS container querie](https://developer.mozilla.org/docs/Web/CSS/Guides/Containment/Container_queries) at MDN.

<!-- ------------------------------ -->
#### Expose unprintable areas via CSS

Use the CSS `page-margin-safety` property to avoid trying to print to unprintable areas of a sheet of paper.

A printer usually has a small area at each of the four edges of a sheet of paper that the printer is not capable of marking reliably, usually due to the printer's paper handling mechanism.

The default page margins are expected to be bigger than these paper-edge areas, but if you set margins in your app, and want to add `@page` margin boxes (such as for custom headers and footers), the `page-margin-safety` property provides a way of detecting where it's safe to print.

See also:
* [Staying within the printable area: the `page-margin-safety` property](https://drafts.csswg.org/css-page-3/#page-margin-safety) in _CSS Paged Media Module Level 3_.

<!-- ------------------------------ -->
#### Media element pseudo-classes

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Author: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging

The following CSS pseudo-classes match `<audio>` and `<video>` elements based on their state:
* `:playing`
* `:paused`
* `:seeking`
* `:buffering`
* `:stalled`
* `:muted`
* `:volume-locked`

See also:
* [`:playing` CSS pseudo-class](https://developer.mozilla.org/docs/Web/CSS/Reference/Selectors/:playing) at MDN.
* [`:paused` CSS pseudo-class](https://developer.mozilla.org/docs/Web/CSS/Reference/Selectors/:paused) at MDN.
* [`:seeking` CSS pseudo-class](https://developer.mozilla.org/docs/Web/CSS/Reference/Selectors/:seeking) at MDN.
* [`:buffering` CSS pseudo-class](https://developer.mozilla.org/docs/Web/CSS/Reference/Selectors/:buffering) at MDN.
* [`:stalled` CSS pseudo-class](https://developer.mozilla.org/docs/Web/CSS/Reference/Selectors/:stalled) at MDN.
* [`:muted` CSS pseudo-class](https://developer.mozilla.org/docs/Web/CSS/Reference/Selectors/:muted) at MDN.
* [`:volume-locked` CSS pseudo-class](https://developer.mozilla.org/docs/Web/CSS/Reference/Selectors/:volume-locked) at MDN.

<!-- ------------------------------ -->
#### CSS `alpha()` function

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Author: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging

The CSS `alpha()` function creates a new color by applying a specified alpha value to an existing color.

For example, `alpha(from red / 0.5)` creates a semi-transparent red color with 50% opacity.

See also:
* [Relative Alpha Colors](https://drafts.csswg.org/css-color-5/#relative-alpha) in _CSS Color Module Level 5_.

<!-- ------------------------------ -->
#### `named-feature()` function for CSS `@supports`

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Author: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging

The `named-feature()` function allows CSS `@supports` rules to query for a small set of specific named features that are not possible to test for by using other `@supports` mechanisms, but which are valuable to test for.

See also:
* [Extensions to the `@supports` rule](https://drafts.csswg.org/css-conditional-5/#at-supports-ext) in _CSS Conditional Rules Module Level 5_.

<!-- ------------------------------ -->
#### `overscroll-behavior: chain`

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Author: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging

The CSS `overscroll-behavior` property now support the new `chain` value.

This property affects two independent effects:
* Scroll propagation.
* Local border effect (such as overscroll stretch).

Currently, `overscroll-0behavior` has 3 values:
* `none:` No propagation, no local effect.
* `auto:` Propagation, local effect.
* `contain:` No propagation, local effect.

This feature adds a new value to complete the set:
* `chain:` Propagation, no local effect.

See also:
* [`overscroll-behavior` CSS property](https://developer.mozilla.org/docs/Web/CSS/Reference/Properties/overscroll-behavior) at MDN.
* [Overscroll Behavior Properties](https://drafts.csswg.org/css-overscroll-1/#overscroll-behavior-properties) in _CSS Overscroll Behavior Module Level 1_.

<!-- ------------------------------ -->
#### `flex-wrap:balance`

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Author: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging

`flex-wrap:balance` allows you to distribute content between each `flex-line`, so that the layout appears more balanced.

This is similar to `text-wrap:balance`.

See also:
* [`flex-wrap` CSS property](https://developer.mozilla.org/docs/Web/CSS/Reference/Properties/flex-wrap) at MDN.
* [Flex Line Wrapping: the `flex-wrap` property](https://drafts.csswg.org/css-flexbox-2/#flex-wrap-property) in _CSS Flexible Box Layout Module Level 2_.

<!-- ------------------------------ -->
#### CSS `frame-sizing` property to responsively size `<iframe>` elements

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Author: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging

Using the CSS `frame-sizing` property, you can seamlessly render an iframe's content without scrollbars, by automatically resizing the iframe to fit its content.

See also:
* [\<iframe\> HTML inline frame element](https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/iframe) at MDN.
* [Responsively-sized iframes: the `frame-sizing` property](https://drafts.csswg.org/css-sizing-4/#responsive-iframes) in _CSS Box Sizing Module Level 4_.

<!-- ====================================================================== -->
## HTML features

The following new HTML features are included in Microsoft Edge.

<!-- ------------------------------ -->
#### `focusgroup` attribute

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Author: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging

Author: Yes, shipping.

```suggestion
```

Additionally, you can now safely nest a `popover=auto` element inside a `popover=hint` element. The nested `popover=auto` will behave as a `popover=hint` element.  This allows use cases such as placing a customizable `<select>` element within a `popover=hint` element.

See also:
* [Popover API](https://developer.mozilla.org/docs/Web/API/Popover_API) at MDN.
* [`popover` HTML global attribute](https://developer.mozilla.org/docs/Web/HTML/Reference/Global_attributes/popover) at MDN.

<!-- ------------------------------ -->
#### Updating all `<selectedcontent>` element in a customizable `<select>`

When multiple `<selectedcontent>` elements are defined in a customizable `<select>` element, all of them are now be kept up to date with the current selected option, instead of only the first one in DOM order.

See also:
* [Customizable select elements](https://developer.mozilla.org/docs/Learn_web_development/Extensions/Forms/Customizable_select) at MDN.
* [\<selectedcontent\> HTML selected option display element](https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/selectedcontent) at MDN.

<!-- ------------------------------ -->
#### Streaming out-of-order HTML

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Author: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging

<!-- ------------------------------ -->
#### Expose the `FontFaceSet` interface globally

Previously, the `FontFaceSet` interface was not exposed as a global property.  Chromium's interface definition language (IDL) for `FontFaceSet` previously incorrectly applied `LegacyNoInterfaceObject` to `FontFaceSet`, hiding it as a global property.

This change aligns Chromium with the CSS Font Loading spec, and with Safari and Firefox.

See also:
* [FontFaceSet](https://developer.mozilla.org/docs/Web/API/FontFaceSet) at MDN.

<!-- ------------------------------ -->
#### Disable SVG filters on cross-origin iframes and plugins

SVG filters are no longer applied to the following:
* Cross-origin or restricted iframes (such as sandboxed iframes).
* Embedded plugins (such as PDFs).

This prevents potential security issues from cross-origin content being processed through SVG filter effects.

See also:
* [SVG filters](https://developer.mozilla.org/docs/Web/SVG/Guides/SVG_filters) at MDN.
* [&lt;iframe&gt; HTML inline frame element](https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/iframe) at MDN.

<!-- ------------------------------ -->
#### IndexedDB: SQLite backend

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Author: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging

In Edge, the implementation of IndexedDB now uses SQLite to improve reliability and, to a lesser extent, performance.

With this change, the first time a user visits a site, or after clearing site data, new data is stored in the IndexedDB backend that uses SQLite.  Existing data stores will be migrated in a future release.

See also:
* [IndexedDB API](https://developer.mozilla.org/docs/Web/API/IndexedDB_API) at MDN.

<!-- ------------------------------ -->
#### `MediaStreamTrackProcessor` frame counters 

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Author: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging

This aligns Edge with the HTML Standard and improves security by isolating these workers from same-origin state, such as `BroadcastChannel`, `localStorage`, and `IndexedDB`.

To preserve isolation boundaries, these workers remain in the same storage partition as their creator (for example, by top-level site or nonce).

See also:
* [RFC 2397: The "data" URL scheme](https://datatracker.ietf.org/doc/html/rfc2397).
* [Step 3 of Script settings for workers](https://html.spec.whatwg.org/multipage/workers.html#script-settings-for-workers) in _HTML Living Standard_.

<!-- ------------------------------ -->
#### PWA origin migration

Web App Origin Migration enables a Progressive Web App (PWA) to migrate to a new origin while preserving trust, install state, and applicable permissions.

This helps teams safely move apps across domains without forcing users to reinstall.

See also:
* [Web App Manifest](https://developer.mozilla.org/docs/Web/Manifest) at MDN.

<!-- ------------------------------ -->
#### Get notified when the `scrollBy` and `scrollTo` methods complete

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Author: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging

Programmatic scrolling methods, such as `scrollBy` and `scrollTo`, now return a `Promise` object that resolves when the scroll completes.  Use this promise to run code after a smooth-scroll finishes, without relying on a timer or scroll-event polling.

See also:
* [Window: scrollBy() method](https://developer.mozilla.org/docs/Web/API/Window/scrollBy) at MDN.
* [Window: scrollTo() method](https://developer.mozilla.org/docs/Web/API/Window/scrollTo) at MDN.

<!-- ------------------------------ -->
#### Web Speech API: on-device recognition quality

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Author: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging

The `SpeechRecognition` interface now supports the `quality` property.

This property allows you to specify the semantic capability that's required for recognition which uses `processLocally: true`, also known as on-device recognition.

The values that are supported by the `quality` property map to increasing task complexity and hardware requirements:
* `command`
* `dictation`
* `conversation`

Using the `quality` property, you can determine if the device can handle your use case by using on-device recognition, or if you should fallback to a cloud recognition service.

See also:
* [SpeechRecognition](https://developer.mozilla.org/docs/Web/API/SpeechRecognition) at MDN.
* [Convert speech to text with the SpeechRecognition API](../speech-recognition-api.md).

#### WebGPU: Immediates

> <!-- todo: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging. -->

Author: feature is not 100% guaranteed to ship in 150. Verify if listed at https://chromestatus.com/roadmap before merging

Author: ```suggestion
```

WebGPU now supports immediates.

This feature adds an `immediate` address space in WGSL and `setImmediateData()` methods on render pass, compute pass, and render bundle encoders.  You can use these APIs to pass small, frequently updated values to shaders without creating GPU buffers or bind groups.

This is useful for per-draw data, such as object indices, material indices, and transform values.

See also:
* [WebGPU API](https://developer.mozilla.org/docs/Web/API/WebGPU_API) at MDN.
* [WebGPU](https://gpuweb.github.io/gpuweb/) specification.
* [WebGPU Shading Language (WGSL)](https://gpuweb.github.io/gpuweb/wgsl/) specification.

<!-- ====================================================================== -->
## Origin trials

The following are origin trials for new experimental APIs that are available in Microsoft Edge.

Origin trials let you try experimental APIs on your own live website for a limited time.  To learn more about origin trials, see [Use origin trials in Microsoft Edge](../../origin-trials/index.md).

For the full list of available origin trials, see [Microsoft Edge Origin Trials](https://developer.microsoft.com/microsoft-edge/origin-trials/).

> todo: compare table rows & their wording in [Microsoft Edge 149 web platform release notes (Jun. 2026)](./149.md)

Author: todo

| Soft Navigation Heuristics | &quot;Soft navigations&quot; are JS-driven same-document navigations that are using the history API or the new Navigation API, triggered by a user gesture and modifies the DOM, modifying the previous content, as well as the URL displayed to the user. This OT would experiment with soft navigation heuristics, and will experimentally web-expose them, so that RUM providers and developers can collect them and report back of their usefulness when collecting performance metrics. | [Register](https://developer.microsoft.com/en-us/microsoft-edge/origin-trials/trials/ec952482-49f0-4980-a451-13ea3f9a7220) |
| WebNN | WebNN provides a high-level abstraction over the ML acceleration capabilities provided by the platform. The goal of this experiment is to understand how well real-world models abstract to the operations supported by WebNN and how well WebNN can map these operations to those supported by the hardware of real-world users. | [Register](https://developer.microsoft.com/en-us/microsoft-edge/origin-trials/trials/19857284-cf52-484b-8b09-8ca50ac9dccb) |

> | `focusgroup` | Provides focus navigation among a set of focusable elements with the arrow keys. | [Register](https://developer.microsoft.com/en-us/microsoft-edge/origin-trials/trials/6d4b86e7-230c-4d23-ab2b-ee8d05f18702) |

Author: If focusgroup is confirmed to be shipping with 150 (see regular section above), then remove this origin trial.

Reviewer: Outdated; [fresh](https://github.com/MicrosoftDocs/edge-developer/pull/3824/files#r3364633335)

---

> title: Microsoft Edge 150 web platform release notes (Jul. 2026)

Reviewer: add . in the script after 3-letter month abbrev variable

@captainbrosset
